sir.c

```diff
@@ -49,56 +49,64 @@ uint16_t sir_geterror(char message[SIR_MAXERROR]) {
     return _sir_geterrcode(_sir_geterror(message));
 }
 
-bool sir_debug(const char* format, ...) {
+PRINTF_FORMAT_ATTR(1, 2)
```

Comment:
Do we really need to add it in the header and source file? Seems like it shouldn't be that way... but what do I know; I've never heard of it.

Reply:
Yes - declaration in the header and the actual function needs to match for it to both work reliably and not give a warning.

Reply:
Okay, what do the 1 and 2 represent?

Reply:
1 is the argument that contains the format string specifier and 2 is the first field which contains the arguments.